### DISCUSSION TITLE
Including Input Widget in SelectionList Selections

### DISCUSSION THREAD

**Discussion by needsleep55** [Textualize/textual#3098] [Q&A] [answered]
Hi,

I'm wondering if it'd be possible to include an Input Widget in an entry of a SelectionList. Something like in HTML with the `<form>` element having `<label>` and `<input>` elements within it.

I had thought Rich's `Group` class could help with this because I thought it was considered a Rich Renderable, but it seems that the Selection class just expects a `TextType` as the prompt.

Any suggestions on how to include an Input Widget next to the prompt in a SelectionList? Or is there a better approach for this?

Thanks

**Comment by davep** [comment, marked as answer]
The SelectionList can't contain other widgets. Depending on what you want to do you may find ListView (which can) is more what you're after.

**Reply by needsleep55** [reply]
I guess as far as presentation that would be a better fit for me, but if I define my own subclass of the ListItem widget for the entries of the ListView, how can I access subclass attributes from the Selected objects passed to on_list_view_selected? Is this possible?

**Reply by davep** [reply]
The objects you're passed will be the objects you create.
